packages/nimble-components/src/table/models/virtualizer.ts

```diff
@@ -122,11 +140,20 @@ export class Virtualizer<TData extends TableRecord = TableRecord> {
 
     private notifyFocusedCellRecycling(): void {
         let tableFocusedElement = this.table.shadowRoot!.activeElement;
+        let parentFocusedElement: Element | undefined;
+        let focusedActionMenuButton: MenuButton | undefined;
         while (
             tableFocusedElement !== null
             && !(tableFocusedElement instanceof TableCellView)
         ) {
+            if (
```

Comment:
Is the `focusedRecycleCallback()` still needed if the cell gets focus on scroll?

Reply:
I guess it's potentially redundant now. It's possible that a column type would want to do something more complicated than just blur in that method (though I can't really think of such a case with our known column types).

We'd also need to make sure the behavior makes sense when the active row is at the top or bottom edge of the viewport (i.e. if the cell isn't visible to focus).

So I probably wouldn't want to remove it in conjunction with this PR, but we should probably evaluate the idea.

Reply:
I played around with your branch a little bit, and I couldn't find a code path where the code within `notifyFocusedCellRecycling` was needed. Specifically, I commented out the whole function and couldn't find anything that broke.

I'm not adamantly opposed to waiting until a follow up PR to remove this code, assuming there was something tracking that work. However, it doesn't seem like the changes made to it in this PR are necessary (unless I'm missing something). I also don't know why a column would ever want or allow a column to do something different when the cell content is blurred based on a scroll/data update vs when the cell content is blurred based on some other keyboard interaction. That makes me think there really isn't a reason why we'd want to keep `focusedRecycleCallback()` around long term.

Reply:
Actually, I think the existing code that focuses the cell on scroll needs to be updated, so it's possible that re-working that code could cause some of the `focusedRecycleCallback` code to still be needed.

The problem is that if you tab away from the table with an action menu open, then scrolling the table with the mouse wheel incorrectly brings focus back into the table when the focus should have stayed on the currently focused item outside of the table.
![TableKeyboardNav](https://github.com/ni/nimble/assets/20542556/c5f4376c-5913-4b73-980a-5016e7ff842c)

Reply:
I pushed a fix for that issue (we'll only refocus the row/cell if focus was within the table before the scroll). [This Storybook](https://60e89457a987cf003efc0a5b-dekebcpwzp.chromatic.com/?path=/story/components-table--table&args=data:LargeDataSet) has the fix.

You're also right that most of the updates to `notifyFocusedCellRecycling()` were unneeded that the cell gets focused too, so I undid most of those.

Reply:
Filed #2202 to track this issue.